pkg/istio-agent/agent.go

```diff
@@ -450,26 +451,32 @@ func (a *Agent) initSdsServer() error {
 // It will not return until both workload cert and root cert are generated.
 //
 // TODO: evaluate replacing the STS server with a file data source, to simplify Envoy config
+// TODO: Fix this method with unused return value
+// nolint: unparam
 func (a *Agent) getWorkloadCerts(st *cache.SecretManagerClient) (sk *security.SecretItem, err error) {
 	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = 0
-
-	for {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Not sure we can feasibly make this change, but I was thinking about this a bit.. its kind of weird that we have a context w/ timeout but also we have MaxElapsedTime - they seem similar purpose. 

Context is still useful if we are canceling for other reason (ctrl+c, etc), but I am not sure if its best to use with timeout?

WDYT? I don't mind either way really but we should probably be consistent/add comments.

Reply:
changed to use MaxElapsedTime to be consistent with other places